gpslogger/src/main/java/com/mendhak/gpslogger/GpsLoggingService.java

```diff
@@ -118,6 +120,16 @@ private void unregisterEventBus(){
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
         super.onStartCommand(intent, flags, startId);
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                startForeground(NOTIFICATION_ID, getNotification(), ServiceInfo.FOREGROUND_SERVICE_TYPE_LOCATION);
+            }
+            else {
+                startForeground(NOTIFICATION_ID, getNotification());
+            }
+        } catch (Exception ex) {
+            LOG.error("Could not start GPSLoggingService in foreground. ", ex);
+        }
```

Comment:
Adding the startForeground into the onStartCommand should make it more reliable anyway. The onCreate only gets called once, but onStartCommand gets called multiple times.  And it should be OK to call startForeground multiple times.  This may improve some reliability perhaps